lib/services/matchers.go

```diff
@@ -57,6 +58,124 @@ func MatchResourceLabels(matchers []ResourceMatcher, resource types.ResourceWith
 	return false
 }
 
+// MatchResourceByFilters returns true if all filter values given matched against the resource.
+func MatchResourceByFilters(resource types.ResourceWithLabels, filter MatchResourceFilter) (bool, error) {
+	if filter.Labels == nil && len(filter.PredicateExpression) == 0 && len(filter.SearchKeywords) == 0 {
+		return true, nil
+	}
+
+	var specResource types.ResourceWithLabels
+
+	// We assume when filtering for services like KubeService, AppServer, and DatabaseServer
+	// the user is wanting to filter the contained resource ie. KubeClusters, Application, and Database.
+	switch filter.ResourceKind {
+	case types.KindNode:
+		specResource = resource
```

Comment:
Do we need to do any type assertions on `resource` here? Or are all resources considered a Node?